agent/engine/docker_task_engine.go

```diff
@@ -332,6 +365,15 @@ func (engine *DockerTaskEngine) synchronizeContainerStatus(container *api.Docker
 		// update the container known status
 		container.Container.SetKnownStatus(currentState)
 	}
+	// Update task ExecutionStoppedAt timestamp
```

Comment:
Please rebase this from dev once #1217 is merged. This is now moved into its own method. May be it's better to review this after that PR is merged